observability/src/test/java/io/grpc/observability/interceptors/LogHelperTest.java

```diff
@@ -0,0 +1,553 @@
+  private final LogSinkWriter sinkWriter =
+      new LogSinkWriter(
+          sink,
+          timeProvider);
+  private final byte[] message = new byte[100];
+
+
+  @Before
+  public void setUp() throws Exception {
+    nonEmptyMetadata.put(KEY_A, DATA_A);
+    nonEmptyMetadata.put(KEY_B, DATA_B);
+    nonEmptyMetadata.put(KEY_C, DATA_C);
+  }
+
+  @Test
+  public void socketToProto_ipv4() throws Exception {
+    InetAddress address = InetAddress.getByName("127.0.0.1");
+    int port = 12345;
+    InetSocketAddress socketAddress = new InetSocketAddress(address, port);
+    assertEquals(
+        Address
+            .newBuilder()
+            .setType(Address.Type.TYPE_IPV4)
+            .setAddress("127.0.0.1")
+            .setIpPort(12345)
+            .build(),
+        LogHelper.socketAddressToProto(socketAddress));
+  }
+
+  @Test
+  public void socketToProto_ipv6() throws Exception {
+    // this is a ipv6 link local address
+    InetAddress address = InetAddress.getByName("2001:db8:0:0:0:0:2:1");
+    int port = 12345;
+    InetSocketAddress socketAddress = new InetSocketAddress(address, port);
+    assertEquals(
+        Address
+            .newBuilder()
+            .setType(Address.Type.TYPE_IPV6)
+            .setAddress("2001:db8::2:1") // RFC 5952 section 4: ipv6 canonical form required
+            .setIpPort(12345)
+            .build(),
+        LogHelper.socketAddressToProto(socketAddress));
+  }
+
+  @Test
+  public void socketToProto_unknown() throws Exception {
+    SocketAddress unknownSocket = new SocketAddress() {
+      @Override
+      public String toString() {
+        return "some-socket-address";
+      }
+    };
+    assertEquals(
+        Address.newBuilder()
+            .setType(Address.Type.TYPE_UNKNOWN)
+            .setAddress("some-socket-address")
+            .build(),
+        LogHelper.socketAddressToProto(unknownSocket));
+  }
+
+  @Test
+  public void metadataToProto_empty() throws Exception {
+    assertEquals(
+        GrpcLogRecord.newBuilder()
+            .setEventType(EventType.GRPC_CALL_REQUEST_HEADER)
+            .setMetadata(
+                GrpcLogRecord.Metadata.getDefaultInstance())
+            .build(),
+        metadataToProtoTestHelper(
+            EventType.GRPC_CALL_REQUEST_HEADER, new Metadata()));
+  }
+
+  @Test
+  public void metadataToProto() throws Exception {
+    assertEquals(
+        GrpcLogRecord.newBuilder()
+            .setEventType(EventType.GRPC_CALL_REQUEST_HEADER)
+            .setMetadata(
+                GrpcLogRecord.Metadata
+                    .newBuilder()
+                    .addEntry(ENTRY_A)
+                    .addEntry(ENTRY_B)
+                    .addEntry(ENTRY_C)
+                    .build())
+            .setPayloadSize(nonEmptyMetadataSize)
+            .build(),
+        metadataToProtoTestHelper(
+            EventType.GRPC_CALL_REQUEST_HEADER, nonEmptyMetadata));
+  }
+
+  @Test
+  public void logRequestHeader() throws Exception {
```

Comment:
If we separate the `LogSinkWriter` as an outer class (its own file) then these tests will also move to  a separate file